### ISSUE TITLE
Per test output when running `go test` is incorrect for parallel tests

### ISSUE THREAD

**Issue by mdwhatcott** [go-lang-plugin-org/go-lang-idea-plugin#1972]
The test runner tool window usually allows you to see output generated by individual tests when you click on the test. However, when tests are marked with `t.Parallel()` (and the -parallel flag has a value greater than 1) then the output is not lined up with each test correctly. That's probably because the output of go test is arranged differently when tests are run in parallel.

Here's a public repository with code you can run that replicates the bug:

https://github.com/mdwhatcott/golang-intellij-testing-in-parallel

**Comment by dlsniper** [comment]
I've managed to reproduce the issue using the sample code. Thank you for the nice report.

**Comment by mdwhatcott** [comment]
You're welcome!

**Comment by zolotov** [comment]
@mdwhatcott thank you for detailed issue. Unfortunately it's not possible (in reasonable time) to decide which test belong which output. https://github.com/golang/go/issues/2981 might change the situation.

**Comment by zolotov** [comment]
I pasted the code in case you want to remove example code from github
```
package testing_stuff

import "testing"

func TestParallel1(t *testing.T) {
	t.Parallel()
	t.Error("1")
}

func TestParallel2(t *testing.T) {
	t.Parallel()
	t.Error("2")
}

func TestParallel3(t *testing.T) {
	t.Parallel()
	t.Error("3")
}

func TestParallel4(t *testing.T) {
	t.Parallel()
	t.Error("4")
}

func TestParallel5(t *testing.T) {
	t.Parallel()
	t.Error("5")
}


func Test1(t *testing.T) {
	t.Error("1")
}

func Test2(t *testing.T) {
	t.Error("2")
}

func Test3(t *testing.T) {
	t.Error("3")
}

func Test4(t *testing.T) {
	t.Error("4")
}

func Test5(t *testing.T) {
	t.Error("5")
}
```

**Comment by mdwhatcott** [comment]
Yeah, I should have just included the code here. Thanks @zolotov.
